test/CMakeLists.txt

```diff
@@ -57,14 +57,19 @@ SET_TARGET_PROPERTIES(
     CatchMain
     PROPERTIES FOLDER "test")
 
-# NVCC does not incorporate the COMPILE_OPTIONS of a target but only the CMAKE_CXX_FLAGS
-IF(ALPAKA_ACC_GPU_CUDA_ENABLE AND NOT ALPAKA_CUDA_COMPILER MATCHES "clang")
+IF(ALPAKA_ACC_GPU_CUDA_ENABLE AND ALPAKA_CUDA_COMPILER MATCHES "nvcc")
+    # NVCC does not incorporate the COMPILE_OPTIONS of a target but only the CMAKE_CXX_FLAGS
     GET_TARGET_PROPERTY(_COMMON_COMPILE_OPTIONS common COMPILE_OPTIONS)
     # If the property does not exist, the variable is set to NOTFOUND.
     IF(_COMMON_COMPILE_OPTIONS)
         STRING(REPLACE ";" " " _COMMON_COMPILE_OPTIONS_STRING "${_COMMON_COMPILE_OPTIONS}")
         SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${_COMMON_COMPILE_OPTIONS_STRING}")
     ENDIF()
+    # nvcc supports werror starting with 10.2
+    IF(CUDA_VERSION GREATER_EQUAL 10.2)
```

Comment:
Adding this here is not ideal either, since for example package managers can run your tests and a simple warning will mark a package with unexpected environment as broken.

Please just enable `-Werror=all-warnings` in CI, e.g. via a travis environment variable. This is a bug :)

Reply:
We already did this for years with the normal compilers Werror flag as most other open source projects are also doing. So this is at least not regression.
Furthermore it is the exact goal of setting the Werror flag to let the tests fail if they do not compile cleanly. Why should we treat warnings differently to compile errors? Having no watnings is our definition of our tests passing. So if a package manager for any reason (I do not know why it should do this at all) runs the tests and they produce a warning on this platform, then our tests fail and this is exactly what we want. That's the whole goal behind Werror.

Reply:
I am just telling you from a package management point of view, that will use software with newer environments than what were available and testable at release, that this is not an ideal practice.

Package managers such as spack, homebrew, et al. run tests to see if a package is working in a larger environment. One usually just runs what comes with the package, to avoid duplication and maintenance overhead.

As developers, we can recommend best practices but it is not our place to anticipate new additions to `-Wall`, `-Wextra` or whatever a compiler decides to warn on. There are more compilers and workflows out there than what we do daily, e.g. people that build their own compiler frontends, optimizer passes, tooling, etc. on projects such as ours. Let's not patronize them.

Reply:
I see your point. However a newer, self built or unknown compiler is in no way different to newer CUDA versions, newer or self built CMake versions, newer OSs or any other dependency. The Compiler is also only a dependency of alpaka. All of those dependencies can make the build or the tests fail at compile or runtime.
I do not see a reason why we should have lower requirements than is possible on the compilation while we expect everything to generate, run and calculate perfectly.
If we lower the requirements on one dependency, we should also reduce the requirements on numerical stability, quality of random numbers and other things that may behave unexpectedly on untested platforms.
Finding issues on untested platforms is the goal of tests and for alpaka a clean compilation is what we defined as a requirement for a successful test execution.

Reply:
> If we lower the requirements on one dependency, we should also reduce the requirements on numerical stability, quality of random numbers and other things that may behave unexpectedly on untested platforms.

I think it's not that black and white. A compiler warning in a newer nvcc might just be fine, no reason to break the tests outside of CI for this.